packages/typescript-checker/src/group.ts

```diff
@@ -0,0 +1,64 @@
+import { Mutant, MutantTestCoverage } from '@stryker-mutator/api/core';
+
+import { SourceFiles } from './compilers/compiler';
+
+import { toPosixFileName } from './fs/tsconfig-helpers';
+
+export function createGroups(sourceFiles: SourceFiles, mutants: MutantTestCoverage[]): MutantTestCoverage[][] {
+  let mutantsWithoutGroup = [...mutants];
+  let groups: MutantTestCoverage[][] = [];
+
+  while (mutantsWithoutGroup.length) {
+    const firstMutant = mutantsWithoutGroup[0];
+    const firstMutantFilename = toPosixFileName(firstMutant.fileName);
+    const firstSourceFile = sourceFiles[firstMutantFilename];
+
+    const group: Array<{ fileName: string; mutant: MutantTestCoverage }> = [{ fileName: firstMutantFilename, mutant: firstMutant }];
+
+    // We allow people to mutate files that are not included in this ts project
+    if (!firstSourceFile) {
+      addGroupToList();
+      continue;
+    }
+
+    let ignoreList = [firstSourceFile.fileName, ...firstSourceFile.importedBy];
```

Comment:
Can be a `const`. Instead of using `ignoreList = [...ignoreList, val, val2]` you can use `ignoreList.push(val, val2)`. Also: you probably want to use a `Set`, because it is much more performant when checking for existence.